packages/ts-moose-lib/src/dmv2/sdk/olapTable.ts

```diff
@@ -210,6 +210,74 @@ export type AggregatingMergeTreeConfig<T> = BaseOlapConfig<T> & {
  */
 export type SummingMergeTreeConfig<T> = BaseOlapConfig<T> & {
   engine: ClickHouseEngines.SummingMergeTree;
+  columns?: string[];
+};
+
+/**
+ * Configuration for ReplicatedMergeTree engine
+ * @template T The data type of the records stored in the table.
+ *
+ * Note: keeperPath and replicaName are optional. Omit them for ClickHouse Cloud,
+ * which manages replication automatically. For self-hosted with ClickHouse Keeper,
+ * provide both parameters or neither (to use server defaults).
+ */
+export type ReplicatedMergeTreeConfig<T> = BaseOlapConfig<T> & {
+  engine: {
+    engine: ClickHouseEngines.ReplicatedMergeTree;
+    keeperPath?: string;
+    replicaName?: string;
+  };
+};
+
+/**
+ * Configuration for ReplicatedReplacingMergeTree engine
+ * @template T The data type of the records stored in the table.
+ *
+ * Note: keeperPath and replicaName are optional. Omit them for ClickHouse Cloud,
+ * which manages replication automatically. For self-hosted with ClickHouse Keeper,
+ * provide both parameters or neither (to use server defaults).
+ */
+export type ReplicatedReplacingMergeTreeConfig<T> = BaseOlapConfig<T> & {
+  engine: {
+    engine: ClickHouseEngines.ReplicatedReplacingMergeTree;
+    keeperPath?: string;
+    replicaName?: string;
+    ver?: keyof T & string;
```

Comment:
constrast this with
```
export type ReplacingMergeTreeConfig<T> = BaseOlapConfig<T> & {
  engine: ClickHouseEngines.ReplacingMergeTree;
  ver?: keyof T & string; // Optional version column
  isDeleted?: keyof T & string; // Optional is_deleted column
};
```

is there a reason why the rest of the config is tucked inside another object?

Reply:
Are you saying extend the other type with the Replicated specifc properties? I can do that